tracer/src/Datadog.Trace/Sampling/ISpanSamplingRule.cs

```diff
@@ -0,0 +1,38 @@
+// <copyright file="ISpanSamplingRule.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Sampling
+{
+    /// <summary>
+    ///     Defines the contract for a sampling rule for single span ingestion.
+    /// </summary>
+    internal interface ISpanSamplingRule
+    {
+        /// <summary>
+        ///     Gets the probability of keeping a span for this rule.
+        /// </summary>
+        /// <value><see langword="float"/> in range <c>[0.0, 1.0]</c>.
+        /// <para><c>0.0</c> is drop all.</para>
+        /// <para><c>1.0</c> is accept all.</para>
+        /// </value>
+        float SamplingRate { get; }
```

Comment:
Any reason for `float` over `double`?

Reply:
Good question
I chose a `float` because the RFC called it a `float`